src/include/daos/mem.h

```diff
@@ -45,6 +45,44 @@
 
 #include <libpmemobj.h>
 
+/** The offset of an object from the base address of the pool */
+typedef uint64_t		umem_off_t;
+#define UMOFF_NUM_FLAGS		(8)
+#define UMOFF_MAX_FLAG		(1ULL << UMOFF_NUM_FLAGS)
+#define UMOFF_FLAG_MASK		(UMOFF_MAX_FLAG - 1)
+#define UMOFF_NULL		(~0ULL & ~UMOFF_FLAG_MASK)
+#define UMOFF_IS_NULL(umoff)	(umoff >= UMOFF_NULL)
```

Comment:
Sorry I'm confused by these macros, could you please explain?

Reply:
I've "reserved" 8 bits when the address is greater than 0xffffffffffffff00 for "NULL" values.  This is to support potential flags such as what DTX currently does with UMMID's for marking a transaction record as "aborted".   The idea is to interpret any such value as NULL but to be able to get the bits to check for secondary values (e.g. NULL, aborted, ...).

Reply:
I'd use 0 for NULL except when used as an offset, I can't guarantee that 0 offset is a NULL pointer.   It seems more likely that 0xffffffffffffff00 is an invalid address even for vmem.

pmemobj_root isn't at 0 in practice so maybe we could actually reserve the first 256 values without any issues as well...that would allow 0 to be NULL

Reply:
If we can use 0 as NULL then we probably should, otherwise it will break all the checks like ```if (!addr)```